mvrx/src/main/kotlin/com/airbnb/mvrx/MvCorStateStore.kt

```diff
@@ -0,0 +1,117 @@
+package com.airbnb.mvrx
+
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.channels.actor
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.rx2.asObservable
+import java.util.*
+import kotlin.coroutines.CoroutineContext
+
+
+/**
+ * This is a container class around the actual state itself. It has a few optimizations to ensure
+ * safe usage of state.
+ *
+ * All state reducers are run in an actor to ensure that they don't have race
+ * conditions with each other.
+ *
+ */
+
+internal open class MvCorStateStore<S : Any>(initialState: S, final override val coroutineContext: CoroutineContext = Dispatchers.Default) : Disposable, CoroutineScope, MvRxStateStore<S> {
```

Comment:
rather than having the class implement `CoroutineScope` could you have it own a coroutine scope as a private property.  Android did a similar thing with fragment.lifecycleScope.  Think of it as not repeating the problem of an activity being a context but rather an activity owning a context.

Reply:
It's just for convenince since every coroutine expects to be started in CoroutineScope, and this class should be passed around by it's interface so it doesn't expose it's scope. Also in [kotlin documentation ](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/) activity implements coroutine scope through delegate. If you want I'll refactor to private property, I don't have a strong opinion on which way is the right one